### ISSUE TITLE
Packaging fails with PyPI sourced pup

### ISSUE THREAD

**Issue by tmontes** [mu-editor/pup#77]
DETAILS:

* Packaging `puppy` is failing with the latest PyPI-sourced `pup` 1.0.0a3 release.

REPRODUCTION:

* Create a Python 3.7 venv and update `pip`.
* `pip install pup` -- grabs 1.0.0a3 from PyPI.
* Clone `puppy` source tree from https://github.com/tmontes/puppy.
* Run `pup package .` on the source tree root.

RESULT (Windows):
```
(venv-xirro) C:\Users\test\work\github.com\puppy>pup package .
I pup 1.0.0a3 - starting with PID=3156
I Package '.': starting.
I Collecting metadata for '.'.
I About to run 'c:\\users\\test\\work\\github.com\\puppy\\venv-xirro\\scripts\\python.exe -m pip wheel --no-deps C:\\Users\\test\\work\\github.com\\puppy'.
I pip out: Processing c:\users\test\work\github.com\puppy
I pip out: Building wheels for collected packages: puppy
I pip out:   Building wheel for puppy (setup.py): started
I pip out:   Building wheel for puppy (setup.py): finished with status 'done'
I pip out:   Created wheel for puppy: filename=puppy-1.4.0-py3-none-any.whl size=8648 sha256=ff3f236e235e73f658ce6ae95e4f0786a4065d7e8515647b21a8f910a5c80590
I pip out:   Stored in directory: C:\Users\test\AppData\Local\Temp\pip-ephem-wheel-cache-vyuiro5s\wheels\17\26\43\705bd58c4b717c1af27e3ec8fb612c8da7e59b082813f42eb8
I pip out: Successfully built puppy
I Step 'win.distribution-layout': starting.
C Execution failure:
C Traceback below:
Traceback (most recent call last):
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\pup\__main__.py", line 31, in wrapper
    exit_code = command_function(*args, **kw)
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\pup\__main__.py", line 80, in package
    launch_module=launch_module,
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\pup\api.py", line 41, in package
    dsp.run_pluggable_step(step)
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\pup\dispatcher.py", line 79, in run_pluggable_step
    return self._invoke_plugin(name)
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\pup\dispatcher.py", line 68, in _invoke_plugin
    return plugin(self._ctx, self, **kwargs)
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\pup\plugins\win\dist_layout.py", line 63, in __call__
    result_path = generate.generate_files(tmpl_path, tmpl_data, build_dir, overwrite_if_exists=True)
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\cookiecutter\generate.py", line 264, in generate_files
    template_dir = find_template(repo_dir)
  File "c:\users\test\work\github.com\puppy\venv-xirro\lib\site-packages\cookiecutter\find.py", line 34, in find_template
    raise NonTemplatedInputDirException
cookiecutter.exceptions.NonTemplatedInputDirException
I pup 1.0.0a3 - done

(venv-xirro) C:\Users\test\work\github.com\puppy>
```

RESULT (macOS):
```
(venv) tiago.montes (~/Temp/i77/puppy): pup package .
I pup 1.0.0a3 - starting with PID=1032
I Package '.': starting.
I Collecting metadata for '.'.
I About to run '/Users/tiago.montes/Temp/i77/venv/bin/python3.7 -m pip wheel --no-deps /Users/tiago.montes/Temp/i77/puppy'.
I pip out: Processing /Users/tiago.montes/Temp/i77/puppy
I pip out: Building wheels for collected packages: puppy
I pip out:   Building wheel for puppy (setup.py): started
I pip out:   Building wheel for puppy (setup.py): finished with status 'done'
I pip out:   Created wheel for puppy: filename=puppy-1.4.0-py3-none-any.whl size=8621 sha256=cea794985820a128e2dfdbccdb480447e05a0f63cfdb04620aea41962d15af88
I pip out:   Stored in directory: /private/var/folders/0m/xy6669f52zdbb6prv46hrghw0000gp/T/pip-ephem-wheel-cache-as8l2gbp/wheels/fe/32/0e/c92f7a965c6b69e2cc9d2076cc0dd7078458303396513efbcd
I pip out: Successfully built puppy
I Step 'mac.app-bundle-template': starting.
C Execution failure: 
C Traceback below:
Traceback (most recent call last):
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/pup/__main__.py", line 31, in wrapper
    exit_code = command_function(*args, **kw)
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/pup/__main__.py", line 80, in package
    launch_module=launch_module,
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/pup/api.py", line 41, in package
    dsp.run_pluggable_step(step)
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/pup/dispatcher.py", line 79, in run_pluggable_step
    return self._invoke_plugin(name)
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/pup/dispatcher.py", line 68, in _invoke_plugin
    return plugin(self._ctx, self, **kwargs)
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/pup/plugins/mac/app_bundle.py", line 65, in __call__
    result_path = generate.generate_files(tmpl_path, tmpl_data, build_dir, overwrite_if_exists=True)
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/cookiecutter/generate.py", line 264, in generate_files
    template_dir = find_template(repo_dir)
  File "/Users/tiago.montes/Temp/i77/venv/lib/python3.7/site-packages/cookiecutter/find.py", line 34, in find_template
    raise NonTemplatedInputDirException
cookiecutter.exceptions.NonTemplatedInputDirException
I pup 1.0.0a3 - done
(venv) tiago.montes (~/Temp/i77/puppy): 
```

HOWEVER:

* It *does* work when running `pup` from its development virtual environment.

THUS:

* We have a meta-packaging issue.
* `pup` is not properly packaged itself. :-)

**Comment by tmontes** [comment]
DIAGNOSTIC:

* The bundled `cookiecutter` templates for macOS application templates and Windows distribution layout aren't being properly packaged in `pup`'s PyPI artifacts.
* Oh the irony!

**Comment by tmontes** [comment]
Fixed with #78, just merged. Closing.
